src/App.tsx

```
@@ -8,53 +8,183 @@ import { PostsList } from './components/PostsList';
-              </div>
+  const [error, setError] = useState(false);
+  const [errorComments, setErrorComments] = useState(false);
+
+  const [selectedUser, setSelectedUser] = useState(false);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+
+  const [openSidebar, setOpenSidebar] = useState('');
+
+  const [comments, setComments] = useState<Comment[]>([]);
+
+  async function fetchCreatComment({
+    name,
+    email,
+    body,
+    postId,
+  }: Omit<Comment, 'id'>) {
+    try {
+      const result = await creatComments({ name, email, body, postId });
+
+      setComments(currentComments => [...currentComments, result]);
+    } catch {}
+  }
+
+  async function fetchPosts(userId: number) {
+    setLoading(true);
+    setError(false);
+
+    if (userId !== selectedUserId) {
+      setOpenSidebar('Close');
+    }
+
+    if (userId) {
+      setSelectedUser(true);
+    }
+
+    try {
+      const results = await getPost(userId);
+
+      setPosts(results);
+    } catch {
+      setError(true);
+      setLoading(false);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  async function fetchComments(postId: number) {
+    setLoadingComments(true);
+    setErrorComments(false);
+
+    try {
+      const results = await getComments(postId);
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+      setComments(results);
+    } catch {
+      setLoadingComments(false);
+      setErrorComments(true);
+    } finally {
+      setLoadingComments(false);
+    }
+  }
+
+  async function fetchDeleteComment(id: number) {
+    setComments(currentComment =>
+      currentComment.filter(comment => comment.id !== id),
+    );
+
+    try {
+      await deleteComment(id);
+    } catch {}
+  }
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      try {
+        const loadingUsers = await getUsers();
+
+        setUsers(loadingUsers);
+      } catch {}
```

useEffect fetchUsers swallow errors (empty catch). If fetching users fails you should at least log the error; optionally set an error state so user can be notified. Keeping empty catches hides problems during load.